specifyweb/frontend/js_src/lib/components/WbActions/WbSave.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+
+import { useBooleanState } from '../../hooks/useBooleanState';
+import { Button } from '../Atoms/Button';
+import { className } from '../Atoms/className';
+import { Dialog } from '../Molecules/Dialog';
+import { commonText } from '../../localization/common';
+import { wbText } from '../../localization/workbench';
+import { loadingBar } from '../Molecules';
+import { Workbench } from '../WorkBench/WbView';
+import { overwriteReadOnly } from '../../utils/types';
+import { ping } from '../../utils/ajax/ping';
+import { Http } from '../../utils/ajax/definitions';
+import { LoadingContext } from '../Core/Contexts';
+
+export function WbSave({
+  workbench,
+  hasUnsavedChanges,
+  checkDeletedFail,
+  onSpreadsheetUpToDate: handleSpreadsheetUpToDate,
+}: {
+  readonly workbench: Workbench;
+  readonly hasUnsavedChanges: boolean;
+  readonly checkDeletedFail: (statusCode: number) => void;
+  readonly onSpreadsheetUpToDate: () => void;
+}): JSX.Element {
+  const [showProgressBar, openProgressBar, closeProgressBar] =
+    useBooleanState();
+  const loading = React.useContext(LoadingContext);
+
+  const handleSave = () => {
+    // Clear validation
+    overwriteReadOnly(workbench.dataset, 'rowresults', null);
+    workbench.validation.stopLiveValidation();
+
+    // Show saving progress bar
+    openProgressBar();
+
+    // Send data
+    loading(
+      ping(`/api/workbench/rows/${workbench.dataset.id}/`, {
+        method: 'PUT',
+        body: workbench.data,
+        expectedErrors: [Http.NO_CONTENT, Http.NOT_FOUND],
+      })
+        .then((status) => checkDeletedFail(status))
+        .then(() => {
+          handleSpreadsheetUpToDate();
+          workbench.cells.cellMeta = [];
+          // TODO: Figure out how to rework searchCells for SettingsChange as input
+          // workbench.utils?.searchCells({ key: 'SettingsChange' });
```

Comment:
What does `searchCells({ key: 'SettingsChange' })` do exactly?

Reply:
The Advanced search dialog lets you change search config (case sensitivity, whether regex is enabled, etc)
By default, it will do a live preview of the effect of your new search config on the search results.
So, it triggers a search after each change you do to the search settings
But, since such search is triggered automatically, it should not move the user's currently focused cell

That's what this code is for (and the associated comments):
https://github.com/specify/specify7/blob/07a1e5b0846469d7e4b3068077039eb313a4f40e/specifyweb/frontend/js_src/lib/components/WorkBench/WbUtils.tsx#L299-L311

and this:
https://github.com/specify/specify7/blob/07a1e5b0846469d7e4b3068077039eb313a4f40e/specifyweb/frontend/js_src/lib/components/WorkBench/WbUtils.tsx#L384C57-L391

(to not move the currently focused cell unless Enter key was pressed - i.e don't move them on SettingsChange)

If it would be less confusing, instead of `{ key: 'SettingsChange' }`, you could change searchCells signature to allow passing `undefined` as the first argument - to signify that the search is initiated by the code, rather than in response to a user event

Reply:
Any ideas how I can pass `searchQueryElement` as a parameter to `searchCells` on SettingsChange without lifting the state too high up?

Reply:
you will have to lift the state up to WbUtilsComponent?
that's not too high

Reply:
It's used in `WbSave.tsx` after saving spreadsheet changes as well. Do we need it there? If not then I can just lift it up to WbUtilsComponent.

Reply:
From `git blame`, it looks like it was added by me in 2021 in this commit:

https://github.com/specify/specify7/commit/f7914eb2389c53399fd5586e5e1102065366f9f3#diff-4b4981bf4a11f984829caa6ab319badcba99b1f2fb4132ca0a9e37413ea7f55cR1880

It might have been added to re-run search based on all the changes you did since last save:

https://github.com/specify/specify7/blob/f7914eb2389c53399fd5586e5e1102065366f9f3/specifyweb/frontend/js_src/lib/wbutils.js#L242-L246

(in cases that you don't have live search enabled, without re-running search on save, you may see stale search results persist even after saving)

Of course, that might no longer be necessary with react - could you verify?

Reply:
Looks like we still need it otherwise searchResults disappear on save. I've lifted `searchRef` up to WbView